pkg/kube/controllers/boshdeployment/reconciler.go

```diff
@@ -135,24 +475,299 @@ func (r *ReconcileBOSHDeployment) Reconcile(request reconcile.Request) (reconcil
 
 			// Create the extended statefulset
 			err := r.client.Create(ctx, &eSts)
 			if err != nil {
 				r.recorder.Event(instance, corev1.EventTypeWarning, "CreateExtendedStatefulSetForDeployment Error", err.Error())
-				r.log.Errorf("Error creating ExtendedStatefulSet %s for deployment manifest %s: %s", eSts.Name, request.NamespacedName, err)
-				return reconcile.Result{}, errors.Wrap(err, "couldn't create an ExtendedStatefulSet for a BOSH Deployment")
+				r.log.Errorf("Error creating ExtendedStatefulSet %s for deployment manifest %s: %s", eSts.Name, instance.GetName(), err)
+				return errors.Wrap(err, "couldn't create an ExtendedStatefulSet for a BOSH Deployment")
 			}
 		}
 	}
 
-	return reconcile.Result{}, nil
+	instance.Status.State = deployingState
+
+	return nil
+}
+
+// actionOnDeploying check out deployment status
+func (r *ReconcileBOSHDeployment) actionOnDeploying(ctx context.Context, instance *bdc.BOSHDeployment, kubeConfigs *bdm.KubeConfig) error {
+	// TODO Check deployment
+	instance.Status.State = deployedState
+
+	return nil
+}
+
+// newExtendedJobTemplateForVariableInterpolation returns a job to interpolate variables
+func (r *ReconcileBOSHDeployment) newExtendedJobTemplateForVariableInterpolation(ctx context.Context, manifest *corev1.Secret, variables []esv1.ExtendedSecret, jobLabels map[string]string, secretLabels map[string]string, namespace string) (*ejv1.ExtendedJob, error) {
+	cmd := []string{"/bin/sh"}
+	args := []string{"-c", "cf-operator variable-interpolation --manifest /var/run/secrets/manifest/manifest.yaml --variables-dir /var/run/secrets/variables | base64 | tr -d '\n' | echo \"{\\\"interpolated-manifest.yaml\\\":\\\"$(</dev/stdin)\\\"}\""}
+
+	volumes := []corev1.Volume{
+		{
+			Name: string(manifest.GetUID()),
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: manifest.GetName(),
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "manifest.yaml",
+							Path: "manifest.yaml",
+						},
+					},
+				},
+			},
+		},
+	}
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      string(manifest.GetUID()),
+			MountPath: "/var/run/secrets/manifest",
+			ReadOnly:  true,
+		},
+	}
+
+	for _, variable := range variables {
+		vars := Variables{}
+		varName := variable.GetLabels()["variableName"]
+
+		secretItems := []corev1.KeyToPath{}
+
+		// Check if variable already exists
+		foundSecret := &corev1.Secret{}
+		err := r.client.Get(ctx, types.NamespacedName{Name: variable.Name, Namespace: variable.Namespace}, foundSecret)
+		if err != nil && apierrors.IsNotFound(err) {
+			r.log.Debugf("Could not find secret '%s'", variable.GetName())
+			return nil, err
+		} else if err != nil {
+			r.log.Errorf("Failed to get secret '%s': %v", variable.GetName(), err)
+			return nil, err
+		}
+
+		// Update secret and relative SecretVolumeSource
+		switch variable.Spec.Type {
```

Comment:
I don't understand this switch, and why we update all the secrets.
Can you please explain?

Reply:
It looks like we're moving from `data` to `stringData` - this shouldn't be required.

Reply:
Reply as above comments: https://github.com/cloudfoundry-incubator/cf-operator/pull/147#discussion_r268016548

Reply:
Yep, thanks for the explanation.
Let's change extended secrets so they create what we need, and then remove all of this.